src/App/Fossa/Main.hs

```diff
@@ -24,43 +21,22 @@ appMain = do
 
   maybeApiKey <- checkAPIKey optAPIKey
 
-  basedir <- validateDir optBasedir
-
-  setCurrentDir basedir
-
   case optCommand of
-    AnalyzeCommand AnalyzeOptions{..} ->
-      if analyzeOutput
-        then analyzeMain logSeverity OutputStdout optProjectName optProjectRevision
-        else case maybeApiKey of
-          Nothing -> die "A FOSSA API key is required to run this command"
-          Just key -> analyzeMain logSeverity (UploadScan optBaseUrl key analyzeMetadata) optProjectName optProjectRevision
+    AnalyzeCommand parsedOpts ->
+      analyzeMain parsedOpts maybeApiKey logSeverity optBaseUrl optProjectName optProjectRevision
```

Comment:
it looks like we're just passing in everything from the `CmdOptions` (now poorly-named -- `GlobalOptions`?) -- would it make it clearer to pass in the entire `GlobalOptions`?

Reply:
Agree with your point but not sure how to proceed.  Let's pair on this.